examples/src/bin/adc.rs

```diff
@@ -19,20 +19,19 @@
 use esp_backtrace as _;
 use esp_hal::{
     analog::adc::{Adc, AdcConfig, Attenuation},
-    clock::ClockControl,
     delay::Delay,
     gpio::Io,
-    peripherals::Peripherals,
     prelude::*,
-    system::SystemControl,
 };
 use esp_println::println;
 
 #[entry]
 fn main() -> ! {
-    let peripherals = Peripherals::take();
-    let system = SystemControl::new(peripherals.SYSTEM);
-    let clocks = ClockControl::boot_defaults(system.clock_control).freeze();
+    let System {
```

Comment:
Not sure we should teach this pattern in the examples 🤔JM2C

Reply:
Not only that, this is way more verbose than what I'd prefer, too :)

Reply:
yeah, I was about to mention this is five LOCs vs three LOCs 😆

Reply:
Counting the import changes I think it was 6 -> 6, but now it's only 2 :)